Gems/Terrain/Assets/Shaders/Terrain/TerrainClipmapDebugPass.azsl

```diff
@@ -0,0 +1,181 @@
+    Sampler m_sampler
+    {
+        MaxAnisotropy = 16;
+        AddressU = Wrap;
+        AddressV = Wrap;
+        AddressW = Wrap;
+    };
+}
+
+struct VSInput
+{
+    uint m_vertexIndex : SV_VertexID;
+};
+
+struct VSOutput
+{
+    float4 m_position : SV_Position;
+    float2 m_uv : UV0;
+};
+
+struct PSOutput
+{
+    float4 m_color : SV_Target0;
+};
+
+float3 UnpackNormal(float2 packedNormal)
+{
+    float z = sqrt(1.0 - packedNormal.x * packedNormal.x - packedNormal.y * packedNormal.y);
+    return float3(packedNormal.xy, z);
+}
+
+// This VS is a slightly modified version of fullscreen pass shader.
+// It uses a single triangle but only covers a fixed area (unrelated to the viewport size)
+// at the bottom-right corner.
+// see also <Atom/Features/PostProcessing/FullscreenVertexUtil.azsli>
+VSOutput MainVS(VSInput vsInput)
+{
+    VSOutput OUT;
+
+    float scaleX = PassSrg::m_clipmapSize / PassSrg::m_viewportSize.x * PassSrg::m_scale;
+    float scaleY = PassSrg::m_clipmapSize / PassSrg::m_viewportSize.y * PassSrg::m_scale;
+
+    float u = float(vsInput.m_vertexIndex & 2);
+    float v = float((vsInput.m_vertexIndex << 1) & 2);
+    float x = u * scaleX + (1.0 - scaleX);
+    float y = -v * scaleY - (1.0 - scaleY);
+    OUT.m_position = float4(x, y, 0, 1);
+    OUT.m_uv = float4(u, v, 0, 1);
+    return OUT;
+}
+
+PSOutput MainPS(VSOutput psInput)
+{
+    PSOutput OUT;
+    float3 sampleCoord = float3(psInput.m_uv, PassSrg::m_clipmapLevel);
+    switch (PassSrg::m_clipmapId)
+    {
+        case 0:
+        {
+            float4 color = PassSrg::m_macroColorClipmaps.Sample(PassSrg::m_sampler, sampleCoord);
+            OUT.m_color = float4(color.rgb * PassSrg::m_brightness, 1.0);
+            break;
+        }
+        case 1:
+        {
+            float2 packedNormal = PassSrg::m_macroNormalClipmaps.Sample(PassSrg::m_sampler, sampleCoord);
+            float3 normal = UnpackNormal(packedNormal);
+            normal = (normal + 1.0) / 2.0; // transform to [0, 1] to be displayed on the screen.
+            OUT.m_color = float4(normal.xyz * PassSrg::m_brightness, 1.0);
+            break;
+        }
+        case 2:
+        {
+            float4 color = PassSrg::m_detailColorClipmaps.Sample(PassSrg::m_sampler, sampleCoord);
+            OUT.m_color = float4(color.rgb * PassSrg::m_brightness, 1.0);
+            break;
+        }
+        case 3:
+        {
+            float2 packedNormal = PassSrg::m_detailNormalClipmaps.Sample(PassSrg::m_sampler, sampleCoord);
+            float3 normal = UnpackNormal(packedNormal);
+            normal = (normal + 1.0) / 2.0; // transform to [0, 1] to be displayed on the screen.
+            OUT.m_color = float4(normal.xyz * PassSrg::m_brightness, 1.0);
+            break;
+        }
+        case 4:
+        {
+            float height = PassSrg::m_detailHeightClipmaps.Sample(PassSrg::m_sampler, sampleCoord);
+            float3 splat = float3(height, height, height) * 0.01 * PassSrg::m_brightness;
```

Comment:
Why is the "0.01" needed?

Reply:
Because height can go over 1 then everything will look the same.
I'll add a comment to explain.

Reply:
Looks like you're using an R32_Float for heights - we don't need that much precision for this, and it shouldn't be possible for heights to go above 1 - this texture really should just be R8_UNORM. It's basically the texture you'd use for POM. Not blocking this review but something we should change soon.

Reply:
I have a ticket LYN-12729 to adjust all the parameters including texture format.